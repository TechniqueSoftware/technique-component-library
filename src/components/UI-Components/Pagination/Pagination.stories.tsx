import * as React from 'react';
import { createCUICStoriesOf } from '../utils/utils';
import { MODULE_NAME } from '../constants';
import Pagination from './Pagination';
import { demoTop101Movies } from './data';

const { useState } = React;

function PaginationStory() {
  const [pageNo, setPageNo] = useState(1);
  const [pageSize, setPageSize] = useState(10);

  const movies =
    demoTop101Movies
      .slice((pageNo - 1) * pageSize)
      .slice(0, pageSize);

  if (!movies.length) setPageNo(1);

  return (
    <>
      <Pagination
        count={demoTop101Movies.length}
        pageNo={pageNo}
        pageSize={pageSize}
        onPageNoChange={setPageNo}
        onPageSizeChange={setPageSize}
      />
      {movies.map(movie => (
        <div key={movie.no}>
          <div style={{ display: 'inline-block', minWidth: 46 }}>No {movie.no}.</div>
          <span>{movie.title} ({movie.year})</span>
        </div>
      ))}
    </>
  );
}

createCUICStoriesOf(MODULE_NAME.TABLES, module)
  .addCUICStory({
    render: () => <PaginationStory/>,
    name: 'Pagination',
    notes: require('./notes.md').default
  });
