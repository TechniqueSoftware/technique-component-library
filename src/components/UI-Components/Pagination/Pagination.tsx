import * as React from 'react';
import TablePagination from '@material-ui/core/TablePagination';

export interface PaginationProps {
  className?: string;
  pageNo: number;
  pageSize: number;
  count: number;
  pageSizes?: number[]; // Default: [5, 10, 25, 50, 100]
  onPageNoChange: (pageNo: number) => void;
  onPageSizeChange: (pageSize: number) => void;
}

const Pagination = (
  {
    className,
    pageNo,
    pageSize,
    pageSizes = [5, 10, 25, 50, 100],
    count,
    onPageNoChange,
    onPageSizeChange,
  }: PaginationProps,
) => {

  const handleChangePage = (event: unknown, newPage: number) => {
    onPageNoChange(newPage + 1);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<any>) => {
    onPageSizeChange(parseInt(event.target.value, 10));
  };

  return (
    <TablePagination
      className={className}
      component="span"
      count={count}
      page={pageNo - 1}
      rowsPerPage={pageSize}
      rowsPerPageOptions={pageSizes}
      onPageChange={handleChangePage}
      onRowsPerPageChange={handleChangeRowsPerPage}
    />
  );
};

export default Pagination;
