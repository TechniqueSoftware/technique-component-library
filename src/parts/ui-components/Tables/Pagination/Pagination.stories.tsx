import * as React from 'react';
import { createCUICStoriesOf } from '../../../../components/UI-Components/utils/utils';
import { MODULE_NAME } from '../../../../components/UI-Components/constants';
import Pagination from '../../../../components/UI-Components/Pagination/Pagination';
import { demoTop101Movies } from '../../../../components/UI-Components/Pagination/data';

const { useState } = React;

function PaginationStory() {
  const [pageNo, setPageNo] = useState(1);
  const [pageSize, setPageSize] = useState(10);

  const movies =
    demoTop101Movies
      .slice((pageNo - 1) * pageSize)
      .slice(0, pageSize);

  if (!movies.length) setPageNo(1);

  return (
    <>
      <Pagination
        count={demoTop101Movies.length}
        pageNo={pageNo}
        pageSize={pageSize}
        onPageNoChange={setPageNo}
        onPageSizeChange={setPageSize}
      />
      {movies.map(movie => (
        <div key={movie.no}>
          <div style={{ display: 'inline-block', minWidth: 46 }}>No {movie.no}.</div>
          <span>{movie.title} ({movie.year})</span>
        </div>
      ))}
    </>
  );
}

createCUICStoriesOf(MODULE_NAME.TABLES, module)
  .addCUICStory({
    render: () => <PaginationStory/>,
    name: 'Pagination',
    notes: require('./notes.md').default
  });
