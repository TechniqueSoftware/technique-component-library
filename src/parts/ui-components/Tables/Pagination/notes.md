# Pagination
This component implements the [Table Pagination](https://material-ui.com/api/table-pagination/) from the material-ui.
It can be used by tables or paginated lists.

## JIRA
Related jira ticket: https://clubos.atlassian.net/browse/CR-1161
