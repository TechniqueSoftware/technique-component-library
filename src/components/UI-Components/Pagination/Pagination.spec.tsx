import * as React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination', () => {
  test('should render with minimal props', () => {
    const wrapper = shallow(
      <Pagination
        pageNo={1}
        pageSize={20}
        count={200}
        onPageNoChange={() => undefined}
        onPageSizeChange={() => undefined}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render with page sizes', () => {
    const wrapper = shallow(
      <Pagination
        pageNo={1}
        pageSize={20}
        count={200}
        pageSizes={[5, 10, 20, 200]}
        onPageNoChange={() => undefined}
        onPageSizeChange={() => undefined}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render with page 5', () => {
    const wrapper = shallow(
      <Pagination
        pageNo={5}
        pageSize={50}
        count={200}
        onPageNoChange={() => undefined}
        onPageSizeChange={() => undefined}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
