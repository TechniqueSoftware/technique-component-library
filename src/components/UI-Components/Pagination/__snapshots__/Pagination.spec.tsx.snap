// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render with minimal props 1`] = `"<WithStyles(ForwardRef(TablePagination)) className={[undefined]} component=\\"span\\" count={200} page={0} rowsPerPage={20} rowsPerPageOptions={{...}} onPageChange={[Function: handleChangePage]} onRowsPerPageChange={[Function: handleChangeRowsPerPage]} />"`;

exports[`Pagination should render with page 5 1`] = `"<WithStyles(ForwardRef(TablePagination)) className={[undefined]} component=\\"span\\" count={200} page={4} rowsPerPage={50} rowsPerPageOptions={{...}} onPageChange={[Function: handleChangePage]} onRowsPerPageChange={[Function: handleChangeRowsPerPage]} />"`;

exports[`Pagination should render with page sizes 1`] = `"<WithStyles(ForwardRef(TablePagination)) className={[undefined]} component=\\"span\\" count={200} page={0} rowsPerPage={20} rowsPerPageOptions={{...}} onPageChange={[Function: handleChangePage]} onRowsPerPageChange={[Function: handleChangeRowsPerPage]} />"`;
